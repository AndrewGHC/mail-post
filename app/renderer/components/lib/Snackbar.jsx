// @flow
import React, { Component } from 'react';
import MaterialSnackbar from 'material-ui/Snackbar';
import IconButton from 'material-ui/IconButton';
import CloseIcon from 'material-ui-icons/Close';

type Props = {
  id: string,
  message: string,
};

type State = {
  open: boolean,
};

export default class Snackbar extends Component<Props, State> {
  state = {
    open: false,
  };

  componentWillReceiveProps(nextProps: Props) {
    const { id } = this.props;
    const nextId = nextProps.id;
    if (id !== nextId) this.setState({ open: true });
  }

  handleClose = () => {
    this.setState({ open: false });
  };

  render() {
    const { message } = this.props;
    const { open } = this.state;
    return (
      <MaterialSnackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'left',
        }}
        open={open}
        autoHideDuration={6000}
        onClose={this.handleClose}
        SnackbarContentProps={{
          'aria-describedby': 'message-id',
        }}
        message={<span id="message-id">{message}</span>}
        action={[
          <IconButton
            key="close"
            aria-label="Close"
            color="inherit"
            onClick={this.handleClose}
          >
            <CloseIcon />
          </IconButton>,
        ]}
      />
    );
  }
}
