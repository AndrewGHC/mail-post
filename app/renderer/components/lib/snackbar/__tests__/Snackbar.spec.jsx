import React from 'react';
import { mount } from 'enzyme';
import IconButton from 'material-ui/IconButton';
import MaterialSnackbar from 'material-ui/Snackbar';

import Snackbar from '../Snackbar';

const props = {
  id: 'an id',
  message: 'hello',
}

describe('Snackbar', () => {
  it('matches snapshot', async () => {
    const wrapper = mount(<Snackbar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('is closed on mount', async () => {
    const wrapper = mount(<Snackbar {...props} />);
    expect(wrapper.find(MaterialSnackbar).prop('open')).toBe(false);
  });

  it('when passed a new message and id, renders the snackbar', async () => {
    const newMessage = {
      id: 'unique',
      message: 'test',
    };
    const wrapper = mount(<Snackbar {...props} />);
    wrapper.setProps(newMessage);
    expect(wrapper.find(MaterialSnackbar).prop('open')).toBe(true);
  });

  it('when close button is clicked, the snackbar closes', async () => {
    const newMessage = {
      id: 'unique',
      message: 'test',
    };
    const wrapper = mount(<Snackbar {...props} />);
    wrapper.setProps(newMessage);
    wrapper.find(IconButton).simulate('click');
    expect(wrapper.find(MaterialSnackbar).prop('open')).toBe(false);
  });
});
