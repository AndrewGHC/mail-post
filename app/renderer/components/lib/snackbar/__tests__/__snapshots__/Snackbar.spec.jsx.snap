// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar matches snapshot 1`] = `
<Snackbar
  id="an id"
  message="hello"
>
  <WithStyles(Snackbar)
    SnackbarContentProps={
      Object {
        "aria-describedby": "message-id",
      }
    }
    action={
      Array [
        <WithStyles(IconButton)
          aria-label="Close"
          color="inherit"
          onClick={[Function]}
        >
          <pure(Close) />
        </WithStyles(IconButton)>,
      ]
    }
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "bottom",
      }
    }
    autoHideDuration={6000}
    message={
      <span
        id="message-id"
      >
        hello
      </span>
    }
    onClose={[Function]}
    open={false}
  >
    <Snackbar
      SnackbarContentProps={
        Object {
          "aria-describedby": "message-id",
        }
      }
      action={
        Array [
          <WithStyles(IconButton)
            aria-label="Close"
            color="inherit"
            onClick={[Function]}
          >
            <pure(Close) />
          </WithStyles(IconButton)>,
        ]
      }
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "bottom",
        }
      }
      autoHideDuration={6000}
      classes={
        Object {
          "anchorOriginBottomCenter": "MuiSnackbar-anchorOriginBottomCenter-3",
          "anchorOriginBottomLeft": "MuiSnackbar-anchorOriginBottomLeft-7",
          "anchorOriginBottomRight": "MuiSnackbar-anchorOriginBottomRight-5",
          "anchorOriginTopCenter": "MuiSnackbar-anchorOriginTopCenter-2",
          "anchorOriginTopLeft": "MuiSnackbar-anchorOriginTopLeft-6",
          "anchorOriginTopRight": "MuiSnackbar-anchorOriginTopRight-4",
          "root": "MuiSnackbar-root-1",
        }
      }
      disableWindowBlurListener={false}
      message={
        <span
          id="message-id"
        >
          hello
        </span>
      }
      onClose={[Function]}
      open={false}
      transition={[Function]}
      transitionDuration={
        Object {
          "enter": 225,
          "exit": 195,
        }
      }
    />
  </WithStyles(Snackbar)>
</Snackbar>
`;
